### DISCUSSION TITLE
"Insufficient memory: Failed to allocate ... bytes in function cv::OutOfMemoryError, alloc.cpp(52)"

### DISCUSSION THREAD

**Discussion by petroschalas** [bonsai-rx/bonsai#1814] [Q&A]
Hello everyone,

I have been using Bonsai to record behavioural videos with a Basler camera, using the Pylon package. Recently, we changed to a higher resolution camera and recording video with 1920x1920 resolution. When I 
start the recording, after 15min I am getting the "Insufficient memory: Failed to allocate 3686400 bytes in function cv::OutOfMemoryError, alloc.cpp(52)(file 'streamgrabberimpl.cpp', line 469)". 

Are there any recommendations on how to fix this? (workflow attached)
Thank you very much 

[memory_allocation_error.zip](https://github.com/bonsai-rx/bonsai/files/15399946/memory_allocation_error.zip)

**Comment by bruno-f-cruz** [comment]
I would bet your PC cant keep up with the online encoding of the video and thus you keep accumulating memory in RAM until it eventually reaches maximum capacity. Try using alternative encoding strategies. Look in the discussion for "FFMPEG".
